File: app-java/src/main/java/science/monke/internal/order/control/OrderIdConverter.java
package science.monke.internal.order.control;

import science.monke.internal.order.entity.OrderId;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter
public class OrderIdConverter implements AttributeConverter<OrderId, String> {

  @Override
  public String convertToDatabaseColumn(final OrderId orderId) {
    return orderId.value();
  }

  @Override
  public OrderId convertToEntityAttribute(final String orderId) {
    return new OrderId(orderId);
  }
}

File: app-kotlin/src/main/kotlin/science/monke/internal/order/control/OrderIdConverter.kt
package science.monke.internal.order.control;

import science.monke.internal.order.entity.OrderId;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter
class OrderIdConverter : AttributeConverter<OrderId, String> {

    @Override
    override fun convertToDatabaseColumn(orderId: OrderId): String {
        return orderId.value
    }

    override fun convertToEntityAttribute(orderId: String): OrderId {
        return OrderId(orderId);
    }
}
